### ISSUE TITLE
Make DNS alias record creation toggleable / optional

### ISSUE THREAD

**Issue by dlpzx** [awslabs/aws-dataall#797]
**Is your idea related to a problem? Please describe.**
data.all assumes that we can create Route 53 records in customer environment. But this a restricted action that is forbidden with SCPs:
 ```
API: route53:GetHostedZone User: aws:sts::111111111111:assumed-role/cdk-hnb659fds-cfn-exec-role-111111111111-eu-central-1/AWSCloudFormation is not authorized to perform: route53:GetHostedZone on resource: arn:aws:route53mhostedzone/XXXXX with an explicit deny in a service control policy. 
```

**Describe the solution you'd like**

I commented out hosted zone requirement in `deploy/stacks/albfront_stack.py` but we need to evaluate the impact of this action. So it first needs investigation, and if it can be avoided, we would like a solution that makes the creation of DNS alias optional.

*P.S. Don't attach files. Please, prefer add code snippets directly in the message body.*

**Comment by lorchda** [comment]
Make `hosted_zone_name` and `hosted_zone_id` configuration parameters optional.

The `ecs_patterns.ApplicationLoadBalancedFargateService` that you are using in https://github.com/awslabs/aws-dataall/blob/main/deploy/stacks/albfront_stack.py does not require the `domain_zone` parameter (optional), so it can be left out.

**Comment by lorchda** [comment]
See description in #812 for more details.

**Comment by lorchda** [comment]
+ documentation update

**Comment by lorchda** [comment]
Documentation update.

**Comment by dlpzx** [comment]
Merged and released with v2.1.0 🚀
